types/react-dom/server.d.ts

```diff
@@ -39,10 +39,8 @@ export interface RenderToPipeableStreamOptions {
 }
 
 export interface PipeableStream {
-    // tslint:disable-next-line:void-return
-    abort(this: void): void;
-    // tslint:disable-next-line:void-return
-    pipe<Writable extends NodeJS.WritableStream>(this: void, destination: Writable): Writable;
+    abort: () => void;
```

Comment:
this doesn't actually provide a `this: void` contextual type. Arrow *values* have `this` set to undefined, but arrow types still have an unspecified `this` parameter.

Here's a test that might work (this is why tests are encouraged on DT):

```
const o: PipeableStream = {
  abort() {
    // $ExpectType void
    this
  }
}
```

Reply:
What user-relevant difference is there between void and undefined for this? What we're interested in is destructuring for which we already have a test.

Reply:
The difference is between `this: void` and no this-parameter at all, which gives you an implicit `this: any` in the above example. 
Can you link me to the existing destructuring test so I can see how it will change?

I think the reason that arrow types don't specify `this: void` (or undefined [1]) is that they're used to type function expressions as well as arrow expressions, and it would possibly break old code to make `this: void` in a function expression.

[1] `this` parameters predate the `undefined` type by almost a year, which is why they typically use `void`. `undefined` is mostly a supertype of `void`, so it's the type I would use if adding this-parameters now.